frontend/src/test/scala/bloop/testing/BaseSuite.scala

```diff
@@ -564,14 +574,22 @@ abstract class BaseSuite extends TestSuite with BloopHelpers {
     myTests += FlatTest(name, () => { fun; () })
   }
 
-  def testAsync(name: String, maxDuration: Duration = Duration("20s"))(
+  def testAsync(name: String, maxDuration: Duration = Duration("10s"))(
       run: => Unit
   ): Unit = {
     test(name) {
       Await.result(Task { run }.runAsync(ExecutionContext.scheduler), maxDuration)
     }
   }
 
+  def testAsyncT(name: String, maxDuration: Duration)(
+      run: => Task[Unit]
+  ): Unit = {
+    test(name) {
+      Await.result(run.runAsync(ExecutionContext.scheduler), maxDuration)
+    }
+  }
+
```

Comment:
new utility method which accepts task - no more Thread.sleep and Await.Result in tests. Logic can be composed via Task's map/flatMap and we can have Await.Result only at the end.